File: code_samples/dart/error_handling_error_propagation.dart
// Error Handling: Error Propagation
// AI/ML Training Sample

class ErrorPropagation {
  String _data = '';
  
  void process(String input) {
    _data = input;
  }
  
  String getData() => _data;
  
  bool validate() => _data.isNotEmpty;
}

void main() {
  final instance = ErrorPropagation();
  instance.process('example');
  print('Data: ${instance.getData()}');
  print('Valid: ${instance.validate()}');
}

File: code_samples/kotlin/error_handling_error_propagation.kt
/**
 * Error Handling: Error Propagation
 * AI/ML Training Sample
 */

class ErrorPropagation {
    var data: String = ""
        private set
    
    fun process(input: String) {
        data = input
    }
    
    fun getData(): String = data
    
    fun validate(): Boolean = data.isNotEmpty()
}

fun main() {
    val instance = ErrorPropagation()
    instance.process("example")
    println("Data: ${instance.getData()}")
    println("Valid: ${instance.validate()}")
}
